src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Object.cs

```diff
@@ -20,7 +20,88 @@ public partial struct JSMarshalerArgument
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public unsafe void ToManaged(out object? value)
         {
-            throw new NotImplementedException();
+            if (slot.Type == MarshalerType.None)
+            {
+                value = default;
+            }
+            else if (slot.Type == MarshalerType.Object)
+            {
+                value = ((GCHandle)slot.GCHandle).Target;
+            }
+            else if (slot.Type == MarshalerType.Boolean)
+            {
+                ToManaged(out bool v);
+                value = v;
+            }
+            else if (slot.Type == MarshalerType.Double)
+            {
+                ToManaged(out double v);
+                value = v;
+            }
+            else if (slot.Type == MarshalerType.JSObject)
+            {
+                ToManaged(out JSObject? val);
+                value = val;
+            }
+            else if (slot.Type == MarshalerType.String)
+            {
+                ToManaged(out string? val);
+                value = val;
+            }
+            else if (slot.Type == MarshalerType.Exception)
+            {
+                ToManaged(out Exception? val);
+                value = val;
+            }
+            else if (slot.Type == MarshalerType.DateTime)
+            {
+                ToManaged(out DateTime? val);
+                value = val;
+            }
+            else if (slot.Type == MarshalerType.JSException)
+            {
+                ToManaged(out Exception? val);
+                value = val;
+            }
+            else if (slot.Type == MarshalerType.Array)
+            {
+                if(slot.ElementType == MarshalerType.Byte)
+                {
+                    ToManaged(out byte[]? val);
+                    value = val;
+                }
+                else if (slot.ElementType == MarshalerType.Double)
+                {
+                    ToManaged(out double[]? val);
+                    value = val;
+                }
+                else if (slot.ElementType == MarshalerType.Int32)
+                {
+                    ToManaged(out int[]? val);
+                    value = val;
+                }
+                else if (slot.ElementType == MarshalerType.Object)
+                {
+                    ToManaged(out object?[]? val);
+                    value = val;
+                }
+                else
+                {
+                    throw new NotImplementedException("ToManaged: " + slot.ElementType+ "[]");
```

Comment:
Shouldn't this be `NotSupportedException`?

Reply:
I don't have strong opinion. It may be implemented in the future releases.